lib/api/authentication/grader.py

```diff
@@ -26,18 +29,32 @@ def authenticate(self, request):
         user = self.authenticate_credentials(token)
 
         # Make sure that remote address matches service address
-        # TODO: we do not know the language, but we expect that all versions of service_url are within the same domain
-        service_url = user._exercise.as_leaf_class().get_service_url('en')
-        ips = get_url_ip_address_list(service_url)
         ip = get_remote_addr(request)
-        if ip not in ips and ip != '127.0.0.1':
-            logger.error(
-                "Request IP does not match exercise service URL: %s not in %s (%s)",
-                ip,
-                ips,
-                service_url,
-            )
-            raise AuthenticationFailed("Client address does not match service address.")
+        if settings.KUBERNETES_MODE:
```

Comment:
Ok, good point - didn't realize other services authenticate using this as well. Is there a way to know which service is sending the request? If not, could do so that even if k8s mode is on and that auth fails, it still goes to the current else branch, and if that succeeds, we ignore the failed k8s auth attempt.

Reply:
This line `service_url = user._exercise.as_leaf_class().get_service_url('en')` gets the service URL of the exercise grader. The **domain of the URL** separates Rubyric and MOOC-Grader from each other, or Acos server or any other exercise service.

Reply:
It is not ideal to hardcode any URLs/domains in the source code. Do we need to add a new value in settings.py for defining the known exercise services and if they are hosted in virtual machines or Kubernetes?